pdns/tcpiohandler.hh

```diff
@@ -176,6 +193,77 @@ public:
       return readn2WithTimeout(d_socket, buffer, bufferSize, readTimeout, totalTimeout);
     }
   }
+
+  /* Tries to read exactly toRead bytes into the buffer, starting at position pos.
+     Updates pos everytime a successful read occurs,
+     throws an std::runtime_error in case of IO error,
+     return Done when toRead bytes have been read, needRead or needWrite if the IO operation
+     would block.
+  */
+  IOState tryRead(std::vector<uint8_t>& buffer, size_t& pos, size_t toRead)
```

Comment:
Wow, looks like the same code as in `dnsdist-tcp.cc`

Reply:
Yes :-) That's why the one in `dnsdist-tcp.cc` has this comment:

> // XXX could probably be implemented as a TCPIOHandler

but I haven't looked into it yet. It would be nice but it can wait a bit.